cfme/infrastructure/virtual_machines.py

```diff
@@ -1666,17 +1688,34 @@ def step(self, *args, **kwargs):
 
 @navigator.register(InfraTemplateCollection, 'TemplatesOnly')
 class TemplatesAll(CFMENavigateStep):
-    VIEW = TemplatesOnlyAllView
-    prerequisite = NavigateToSibling('All')
+
+    def prerequisite(self):
+        from cfme.infrastructure.provider import InfraProvider
+        if isinstance(self.obj.parent, InfraProvider):
```

Comment:
I'm a bit concerned about this import/comparison, I think it could be done without actually importing these classes.

Can you check and verify that this could be achieved by comparing like:

```
if self.obj.parent is self.obj.provider:
    nav_inst = self.obj.parent
    nav_dest = 'Details'
elif self.obj.parent is self.obj.appliance:
    nav_inst = self.obj
    nav_dest = 'All'
# things blow up if nav_dest or nav_inst isn't set, maybe look for them in locals() or init them
return navigate_to(nav_inst, nav_dest)

```

Reply:
At a minimum, I'd want the import of IPAppliance moved to this local scope, just like InfraProvider, to avoid circular import potential.

Reply:
I implemented something similar that removes the need to import.

Reply:
I implemented NAV_STRING as a property in the Collection classes and had it pass the appropriate string based upon the existence of  a provider filter. Removed this logic from the navigation step class.